src/Makefile

```diff
@@ -0,0 +1,24 @@
+include testing/make_common.mk
+
+# Each subdirectory that has a makefile
+dirs:=$(dir $(wildcard ./*/Makefile))
+
+.PHONY: test
+test: ## Run all tests
+	$(call FOREACH_MAKE,$@,$(dirs))
+
+.PHONY: smoke-test
+smoke-test: ## Run all "smoke tests", establishing a baseline of sanity across the project
+	$(call FOREACH_MAKE_OPTIONAL,$@,$(dirs))
+
+.PHONY: spec-test
+spec-test: ## Run all "specification tests", validating this implementation against the Metaschema spec
```

Comment:
Blocking: is this intended to be incomplete at this time and to be improved later?

```sh
$ make spec-test
Running makefile target 'spec-test' on all subdirectory makefiles that contain the rule
Makefile target 'spec-test' does not exist in ./document/. Continuing...
```

Reply:
This is intended to be just a prototype, the full implementation is tracked via https://github.com/usnistgov/metaschema-xslt/issues/40